### PR TITLE
PLT-3087 Added additional segment events

### PR DESCRIPTION
**Pull request by DavidLu1997**
Changed `updatedUser` and others to add new segment events.

Merge before https://github.com/mattermost/platform/pull/3495

### CODE REVIEW

File: client.jsx [hunk 1]
```diff
@@ -613,7 +613,11 @@ export default class Client {
             send(user).
             end(this.handleResponse.bind(this, 'updateUser', success, error));
 
-        this.track('api', 'api_users_update');
+        if (type === '') {
```
Comment (jwilander) [inline client.jsx]: Might be better to do
```
if (type) {
    this.track('api', 'api_users_update_' + type);
} else {
    this.track('api', 'api_users_update');
}
```

That way it handles `type = null` as well

**Comment by DavidLu1997** [comment]
@jwilander changed

**Comment by coreyhulen** [comment]
+1

**Comment by enahum** [comment]
@DavidLu1997 does this need more work? if not +1

**Comment by DavidLu1997** [comment]
@enahum nope, it is done, I need to submit the platform PR afterwards though

**Pull request merged**